### PR TITLE
MM-49368 - When pasting a link from github it clears the undo history

### LINKED ISSUE #22039: When pasting a link from github it clears the undo history
### Summary

When pasting a link from github with `cmd<ins></ins>shift<ins></ins>c`
  1. Go into Mattermost and paste the link with `cmd+v`
  1. try to undo this with `cmd_z`, no undo.

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-49368

### PR DESCRIPTION
**Pull request by gitstart**
### Demo video
https://www.loom.com/share/5bb0587f30964a84b3a3cd44916b7c10

### Release note
```release-note
Fixes undo history bug when pasting a link from GitHub
```

---
This code was written and reviewed by GitStart Community. Growing great engineers, one PR at a time.

### CODE REVIEW

File: components/advanced_create_post/advanced_create_post.tsx [hunk 1]
```diff
@@ -931,6 +931,7 @@ class AdvancedCreatePost extends React.PureComponent<Props, State> {
         const originalSize = message.length;
         const formattedMessage = formatMarkdownMessage(clipboardData, message.trim(), this.state.caretPosition);
         const newCaretPosition = formattedMessage.length - (originalSize - this.state.caretPosition);
+        document.execCommand('insertText', false, formattedMessage);
```
Comment (gitstart) [inline components/advanced_create_post/advanced_create_post.tsx:934]: From docs

Screenshot 2023-02-14 at 18 12 59

**Review by AshishDhama — APPROVED** [review]

**Review by ghost — APPROVED** [review]
LGTM!

**Review by hmhealey — CHANGES_REQUESTED** [review]
Thanks for the PR! I have some feedback about how to change this, but I like the approach used here.

Also, this component is only used in the centre channel, but not when replying from the RHS. Could you also make the same changes to `AdvancedCreateComment` as well so that it has the same behaviour?

**Comment by gitstart** [comment]
@hmhealey feedback has been addressed.

**Review by hmhealey — CHANGES_REQUESTED** [review]
Thanks for updating all those tests!

Could you also address 
> Also, this component is only used in the centre channel, but not when replying from the RHS. Could you also make the same changes to `AdvancedCreateComment` as well so that it has the same behaviour?

**Comment by gitstart** [comment]
@hmhealey changes have been made to `AdvancedCreateComment`

**Review by hmhealey — APPROVED** [review]
Thanks for adding that to the RHS. I've gone through and double-checked that it works in the centre channel, RHS, and Threads view in Firefox, Chrome, and Safari on Mac

**Comment by M-ZubairAhmed** [comment]
Hey there!, Thank you for your PR.
Webapp is now part of mattermost monorepo now at https://github.com/mattermost/mattermost-server/tree/master/webapp Would you be interested in opening the PR over there?

**Comment by hmhealey** [comment]
The web app now lives in the `webapp/channels` over in https://github.com/mattermost/mattermost-server. We decided to move it to be a monorepo to make both development and our CI pipelines easier, but it's unfortunately disruptive right now. We've updated [our setup docs](https://developers.mattermost.com/contribute/developer-setup/) and added [some steps for migrating to working with the monorepo](https://developers.mattermost.com/contribute/monorepo-migration-notes/).

Copying the changes over there hopefully shouldn't be too hard. There's a few changes to file paths (such as the code living in a dedicated `src` subdirectory), but most things should look the same as they do now. 

Let us know if you could use any help with this. Hopefully the process is smooth.

**Comment by gitstart** [comment]
@hmhealey This PR has been moved  to the monorepo https://github.com/mattermost/mattermost-server/pull/22719 cc @M-ZubairAhmed

**Pull request closed without merging**